x-pack/plugins/security/public/management/roles/edit_role/validate_role.ts

```diff
@@ -72,7 +76,17 @@ export class RoleValidator {
         )
       );
     }
-    if (!role.name.match(NAME_REGEX)) {
+    if (this.buildFlavor === 'serverless' && !role.name.match(SERVERLESS_NAME_REGEX)) {
+      return invalid(
+        i18n.translate(
+          'xpack.security.management.editRole.validateRole.serverlessNameAllowedCharactersWarningMessage',
+          {
+            defaultMessage:
+              'Name must contain only letters, numbers and trailing dots, hyphens or underscores.',
```

Comment:
Not sure of the best wording here.
```suggestion
              'Name must contain only alphanumeric characters, and non-leading dots, hyphens, or underscores.',
```